src/rulesets/rest/2022-05-25/specification-rules.ts

```diff
@@ -1,8 +1,45 @@
 import { RuleError, Ruleset, SpecificationRule } from "@useoptic/rulesets-base";
-import { pascalCase } from "change-case";
+import { dotCase, pascalCase, snakeCase } from "change-case";
 import { links } from "../../../docs";
 import { stabilityKey } from "./constants";
 
+interface componentName {
+  localName: string;
+  localProp?: string;
+  ns?: string;
+}
+
+/**
+ * Decode the component name from a possibly namespace and property-qualified OpenAPI component name.
+ *
+ * @param name An OpenAPI component name
+ * @returns Tuple containing the local name (without namespace) and its namespace prefix (undefined if not namespaced).
+ */
+const decodeComponentName = (name: string): componentName => {
+  const parts = name.split(".");
+  if (parts.length > 1) {
+    const lastPart = parts[parts.length - 1];
+    if (lastPart === snakeCase(lastPart)) {
+      // Component is derived from a Typespec model property, of the form namespace.ModelName.property_name.
```

Comment:
Is this formally defined or just a convention? Can we rely on this if multiple services are using different typespec versions?

Reply:
It's a convention. Typespec just concatenates the namespace, model name and property name with '.'.

So this is probably worth more documentation about how to name things in Typespec to make the linter happy. Should I add a section to the docs?

Reply:
Updated the docs with an example, hopefully this makes the intent clearer.

Reply:
Thanks, I was more worried that if it is not formally defined somewhere then a change in the typespec tooling would start generating names in a different format. However looking at the way the names are generated that is unlikely